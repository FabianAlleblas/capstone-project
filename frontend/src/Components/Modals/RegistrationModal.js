import PropTypes from 'prop-types'
import styled from 'styled-components'
import Button from '../Buttons/Button'

RegistrationModal.propTypes = {
  className: PropTypes.string,
  userRegistration: PropTypes.func.isRequired,
  setIsRegistration: PropTypes.func.isRequired,
  formData: PropTypes.object.isRequired,
}

export default function RegistrationModal({
  className,
  userRegistration,
  setIsRegistration,
  formData,
}) {
  return (
    <Wrapper className={className}>
      <Modal>
        <Heading>Please confirm your registration!</Heading>
        <Text>
          Keep in mind, this app is just a reminding tool. Check your plant and
          your soil frequently and adjust the intervals.
          <br />
          <br />
          This app is not responsible for under or overwatering your plant.
          <br />
          <br />
          <br />
          For demonstration purpose the max amount of plants is limited to 6 per
          user!
        </Text>
        <ButtonWrapper>
          <ModalButton onClick={handleRegistration}>Confirm</ModalButton>
          <ModalButton secondaryStyle onClick={closeModal}>
            Cancel
          </ModalButton>
        </ButtonWrapper>
      </Modal>
    </Wrapper>
  )

  function handleRegistration() {
    userRegistration(formData)
    setIsRegistration(false)
  }

  function closeModal() {
    setIsRegistration(false)
  }
}

const Wrapper = styled.div`
  background-color: rgba(0, 0, 0, 0.4);
  display: grid;
  place-items: center;
  width: 100%;
  z-index: 10;
`

const Modal = styled.div`
  background-color: var(--primary-dark);
  border-radius: 20px;
  box-shadow: 4px 4px 10px var(--shadow);
  display: grid;
  gap: 24px;
  padding: 20px;
  place-items: center;
  width: 90%;
`

const Heading = styled.h2`
  color: var(--form-font-color);
  font-size: 1.5rem;
  text-align: center;
`

const Text = styled.p`
  color: var(--form-font-color);
  font-size: 1.1rem;
`

const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  width: 100%;
`

const ModalButton = styled(Button)`
  width: 50%;
  margin: 4px;
`
