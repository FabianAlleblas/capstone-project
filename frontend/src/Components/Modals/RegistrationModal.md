```js
<RegistrationModal
  userRegistration={() => {}}
  setIsRegistration={() => {}}
  formData={{}}
/>
```
