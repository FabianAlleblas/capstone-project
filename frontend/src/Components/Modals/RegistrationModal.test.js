import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import 'jest-styled-components'
import RegistrationModal from './RegistrationModal'

const userRegistrationMock = jest.fn()
const setIsRegistrationMock = jest.fn()
const formData = { email: 'admin@admin.com', password: '123adminlol' }

describe('RegistrationModal', () => {
  it('renders correctly', () => {
    const { container } = render(
      <RegistrationModal
        userRegistration={userRegistrationMock}
        setIsRegistration={setIsRegistrationMock}
        formData={formData}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls userRegistrationMock and setIsRegistrationMock with the correct data on confirm', () => {
    const { getByText } = render(
      <RegistrationModal
        userRegistration={userRegistrationMock}
        setIsRegistration={setIsRegistrationMock}
        formData={formData}
      />
    )
    user.click(getByText('Confirm'))

    expect(userRegistrationMock).toHaveBeenCalledWith(formData)
    expect(setIsRegistrationMock).toHaveBeenCalledWith(false)
  })
  it('calls setIsRegistrationMock with the correct data on cancel', () => {
    const { getByText } = render(
      <RegistrationModal
        userRegistration={userRegistrationMock}
        setIsRegistration={setIsRegistrationMock}
        formData={formData}
      />
    )
    user.click(getByText('Cancel'))

    expect(setIsRegistrationMock).toHaveBeenCalledWith(false)
  })
})
