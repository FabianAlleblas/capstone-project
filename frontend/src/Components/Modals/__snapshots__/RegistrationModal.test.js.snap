// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationModal renders correctly 1`] = `
.c5 {
  background-color: var(--primary-light);
  border-radius: 22px;
  border: none;
  box-shadow: 0 0 4px var(--shadow);
  color: var(--primary-dark);
  font-family: 'Josefin sans';
  font-size: 1.25rem;
  font-weight: 700;
  padding: 12px 16px 8px;
  width: 100%;
}

.c7 {
  background-color: var(--primary-dark);
  border-radius: 22px;
  border: 4px solid var(--primary-light);
  box-shadow: 0 0 4px var(--shadow);
  color: var(--primary-light);
  font-family: 'Josefin sans';
  font-size: 1.25rem;
  font-weight: 700;
  padding: 8px 12px 4px;
  width: 100%;
}

.c0 {
  background-color: rgba(0,0,0,0.4);
  display: grid;
  place-items: center;
  width: 100%;
  z-index: 10;
}

.c1 {
  background-color: var(--primary-dark);
  border-radius: 20px;
  box-shadow: 4px 4px 10px var(--shadow);
  display: grid;
  gap: 24px;
  padding: 20px;
  place-items: center;
  width: 90%;
}

.c2 {
  color: var(--form-font-color);
  font-size: 1.5rem;
  text-align: center;
}

.c3 {
  color: var(--form-font-color);
  font-size: 1.1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
}

.c6 {
  width: 50%;
  margin: 4px;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Please confirm your registration!
    </h2>
    <p
      class="c3"
    >
      Keep in mind, this app is just a reminding tool. Check your plant and your soil frequently and adjust the intervals.
      <br />
      <br />
      This app is not responsible for under or overwatering your plant.
      <br />
      <br />
      Optimized for mobile view
      <br />
      (iPhone 6/7/8)
      <br />
      <br />
      For demonstration purpose the max amount of plants is limited to 6 per user!
    </p>
    <div
      class="c4"
    >
      <button
        class="c5 c6"
      >
        Confirm
      </button>
      <button
        class="c7 c6"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</section>
`;
